### DISCUSSION TITLE
MRTKRayInteractor handedness

### DISCUSSION THREAD

**Discussion by j-polden** [microsoft/MixedRealityToolkit-Unity#11821] [Q&A] [answered]
Hi Everyone, 

I have a reference to a MRTKRayInteractor in a monobehaviour script I am writing. I am trying to figure out an elegant way to discern which hand the MRTKRayInteractor belongs to. I saw that the [MRTKRayInteractor ](https://learn.microsoft.com/en-us/dotnet/api/mixedreality.toolkit.input.mrtkrayinteractor?view=mrtkinput-3.1) class has a [IHandedInteractor.Handedness](https://learn.microsoft.com/en-us/dotnet/api/mixedreality.toolkit.input.mrtkrayinteractor.mixedreality-toolkit-ihandedinteractor-handedness?view=mrtkinput-3.1#mixedreality-toolkit-input-mrtkrayinteractor-mixedreality-toolkit-ihandedinteractor-handedness) property, but it is protected and I cant access it. 

Does anyone have a method which could be used? 

Thanks!

**Comment by keveleigh** [comment, marked as answer]
You have two main options, both involving casting. The first directly accesses the property you asked about, while the second is essentially a reimplementation of that property. I'd probably recommend the first!

1. `Handedness handedness = ((IHandedInteractor)rayInteractor).Handedness;`
2. `Handedness handedness = rayInteractor.xrController is ArticulatedHandController articulatedHandController ? articulatedHandController.HandNode.ToHandedness() : Handedness.None;`

**Reply by j-polden** [reply]
First option worked! Thanks a lot :)
